pkg/alert/management/management_controller_test.go

```diff
@@ -0,0 +1,272 @@
+package management_test
+
+import (
+	"context"
+	"testing"
+
+	osmv1 "github.com/openshift/api/monitoring/v1"
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	alertmanagement "github.com/openshift/cluster-monitoring-operator/pkg/alert/management"
+	"github.com/openshift/cluster-monitoring-operator/pkg/client"
+)
+
+const (
+	testNamespace = "openshift-monitoring"
+	testAlertName = "TestAlert"
+)
+
+func TestNewManagementController(t *testing.T) {
+	tests := []struct {
+		name        string
+		client      *client.Client
+		serverAddr  string
+		expectError bool
+		errorMsg    string
+	}{
+		{
+			name:        "success",
+			client:      &client.Client{},
+			serverAddr:  ":8080",
+			expectError: false,
+		},
+		{
+			name:        "nil client",
+			client:      nil,
+			serverAddr:  ":8080",
+			expectError: true,
+			errorMsg:    "client cannot be nil",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
```

Comment:
Use t.Contexht()

Reply:
stdversion: testing.Context requires go1.24 or later (file is go1.23)